src/js/vic-v2/helpers.jsx

```diff
@@ -3,36 +3,67 @@ import Raven from 'raven-js';
 import environment from '../common/helpers/environment.js';
 import { transformForSubmit } from '../common/schemaform/helpers';
 
-export function prefillTransformer(pages, formData, metadata) {
+export function prefillTransformer(pages, formData, metadata, state) {
+  let newPages = pages;
+  let newData = formData;
+
   if (formData && formData.serviceBranches) {
     // Mostly we'll be getting branch lists of one or two branches, creating a Set seems like overkill
     const allowedBranches = pages.veteranInformation.schema.properties.serviceBranch.enum;
     const validUserBranches = formData.serviceBranches.filter(branch => allowedBranches.includes(branch));
 
-    const newData = _.unset('serviceBranches', formData);
+    newData = _.unset('serviceBranches', newData);
     if (validUserBranches.length > 0) {
       newData.serviceBranch = validUserBranches[0];
-      return {
-        metadata,
-        formData: newData,
-        pages: _.set('veteranInformation.schema.properties.serviceBranch.enum', validUserBranches, pages)
-      };
+      newPages = _.set('veteranInformation.schema.properties.serviceBranch.enum', validUserBranches, pages);
     }
+  }
 
-    return {
-      metadata,
-      formData: newData,
-      pages
+  if (state.user.profile.services.includes('identity-proofed')) {
+    newData = _.set('processAsIdProofed', true, newData);
+    newData.originalUser = {
+      veteranSocialSecurityNumber: newData.veteranSocialSecurityNumber,
+      veteranFullName: newData.veteranFullName
     };
   }
 
   return {
     metadata,
-    formData,
-    pages
+    formData: newData,
+    pages: newPages
   };
 }
 
+export function identityMatchesPrefill(formData) {
+  const { originalUser = {} } = formData;
+  return formData.veteranSocialSecurityNumber === originalUser.veteranSocialSecurityNumber
+    && formData.veteranFullName.first === originalUser.veteranFullName.first
+    && formData.veteranFullName.middle === originalUser.veteranFullName.middle
+    && formData.veteranFullName.last === originalUser.veteranFullName.last
+    && formData.veteranFullName.suffix === originalUser.veteranFullName.suffix
+    && formData.veteranDateOfBirth === originalUser.veteranDateOfBirth;
+}
+
+export function transform(form, formConfig) {
+  let newData = _.omit(['verified', 'originalUser', 'processAsIdProofed'], form.data);
+
+  // If we pulled their id info at the start and they haven't changed it, then we can submit on the
+  // backend with id info from MVI and discharge status from eMIS
+  // If they changed it, then we have to verify they're not trying to submit a fradulent
+  // request and process them as an anonymous request
```

Comment:
What's the significance of processing as an anonymous request? This changes the way its handled by Salesforce?

Reply:
Yes, it means that admins in SF will need to review their discharge documentation and verify their identity.